From nobody Mon Sep 17 00:00:00 2001
From: Julian Bradfield <jcb+luu@inf.ed.ac.uk>
Date: Wed Mar 1 10:19:44 2006 +0000
Subject: [PATCH 59/81] USB: PL2303 and TIOCMIWAIT
Patch-mainline: 2.6.17

A while ago, I posted about TIOCMIWAIT not working with the PL2303
USB-serial adapter.

After a brief exchange with Greg, I tracked this to a missing wake-up
in the USB interrupt procedures. I got our systems staff to install
the enclosed very simple patch to our 2.6.12 kernels, and it all works
fine as expected. I guess this should also apply to the latest version
and go into the mainstream.

Apologies for the long delay in posting the result.

The routine being patched is pl2303_update_line_status

Signed-off-by: Julian Bradfield <jcb+luu@inf.ed.ac.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---

 drivers/usb/serial/pl2303.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

96050b11f690878c19c1cd39970ab7325d91e18b
diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index cf2213b..fb3b6b7 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -826,6 +826,7 @@ static void pl2303_update_line_status(st
 	spin_lock_irqsave(&priv->lock, flags);
 	priv->line_status = data[status_idx];
 	spin_unlock_irqrestore(&priv->lock, flags);
+	wake_up_interruptible (&priv->delta_msr_wait);
 
 exit:
 	return;
-- 
1.2.4

